src/ConnectedMode.UnitTests/ServerSentEvents/SSESessionManagerTests.cs

```diff
@@ -0,0 +1,235 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.ConnectedMode.ServerSentEvents;
+using SonarLint.VisualStudio.Core.Binding;
+using SonarLint.VisualStudio.Integration.UnitTests;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UnitTests.ServerSentEvents;
+
+[TestClass]
+public class SSESessionManagerTests
+{
+    [TestMethod]
+    public void MefCtor_CheckIsExported()
+    {
+        MefTestHelpers.CheckTypeCanBeImported<SSESessionManager, ISSESessionManager>(
+            MefTestHelpers.CreateExport<IActiveSolutionBoundTracker>(),
+            MefTestHelpers.CreateExport<ISSESessionFactory>());
+    }
+
+    [TestMethod]
+    public void Ctor_SubscribesToBindingChangedEvent()
+    {
+        var solutionTrackerMock = new Mock<IActiveSolutionBoundTracker>();
+
+        var _ = new SSESessionManager(solutionTrackerMock.Object, Mock.Of<ISSESessionFactory>());
+
+        solutionTrackerMock.VerifyAdd(
+            solutionTracker => solutionTracker.SolutionBindingChanged +=
+                It.IsAny<EventHandler<ActiveSolutionBindingEventArgs>>(), Times.Once);
+    }
+
+    [DataTestMethod]
+    [DataRow(true)]
+    [DataRow(false)]
+    public void OnSolutionChanged_CorrectlyIdentifiesOpenAndClose(bool switchToConnectedMode)
+    {
+        var testScope = new TestScope();
+
+
+        if (switchToConnectedMode)
```

Comment:
The word "switch" implies that there was a previous state and now you're switching to the new one. This is not the case. It's simply "isInConnectedMode", with the methods being RaiseBindingChangedToConnectedMode and RaiseBindingChangedToStandaloneMode